### PR TITLE
Fixes 120 Synchronize data display unit with axis display unit

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

**Comment by msevestre** [comment]
ALso found an issue where the LegendIndex was not set propertly (if the number of curves was inferior to the max LegendIndex, new curves were added at the top!)

File: src/OSPSuite.Presentation/Presenters/Charts/ChartEditorPresenter.cs [hunk 1]
```diff
@@ -571,6 +571,7 @@ public void Handle(ChartUpdatedEvent chartUpdatedEvent)
 
       private void refresh()
       {
+         Chart.SynchronizeDataDisplayUnit();
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/Charts/ChartEditorPresenter.cs]: Synchronization of data and unit is done upon refresh in the ChartEditorPresenter

File: src/OSPSuite.Presentation/Presenters/Charts/CurveSettingsPresenter.cs [hunk 2]
```diff
@@ -72,6 +72,7 @@ public CurveSettingsPresenter(ICurveSettingsView view, IDimensionFactory dimensi
       public void MoveSeriesInLegend(CurveDTO curveBeingMoved, CurveDTO targetCurve)
       {
          _chart.MoveSeriesInLegend(curveBeingMoved.Curve, targetCurve.Curve);
+         CurvePropertyChanged(targetCurve.Curve);
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Presenters/Charts/CurveSettingsPresenter.cs]: we might have to revisit that to be more consistent

File: src/OSPSuite.Presentation/Views/Charts/IChartDisplayView.cs [hunk 3]
```diff
@@ -14,25 +14,8 @@ public interface IChartDisplayView : IView<IChartDisplayPresenter>, IBatchUpdata
       /// </summary>
       Action<int> HotTracked { set; }
 
-      string Title { get; set; }
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Views/Charts/IChartDisplayView.cs]: This is all binding stuffs. No need for properties

File: src/OSPSuite.Presentation/Views/Charts/IChartDisplayView.cs [hunk 4]
```diff
@@ -14,25 +14,8 @@ public interface IChartDisplayView : IView<IChartDisplayPresenter>, IBatchUpdata
...
-      ///    Forces the control to invalidate its client area
-      ///    and immediately redraw itself and any child controls. (Windows.Forms.Control)
-      /// </summary>
-      void Refresh();
-
-      /// <summary>
-      ///    This Method is for internal use only and should not be used by an application directly.
-      ///    Reloads data from the underlying data source and repaints the diagram area.
-      /// </summary>
-      void RefreshData();
```
Comment (msevestre) [inline src/OSPSuite.Presentation/Views/Charts/IChartDisplayView.cs]: I hope we don't need those internal stuff with our new design

File: src/OSPSuite.Core/Chart/CurveChart.cs [hunk 5]
```diff
@@ -156,12 +159,21 @@ public void AddCurve(Curve curve, bool useAxisDefault = true)
          _curves.Add(curve);
 
          if (!curve.LegendIndex.HasValue)
-            curve.LegendIndex = Curves.Count;
+            curve.LegendIndex = nextLegenIndex();
+      }
+
+      private int? nextLegenIndex()
```
Comment (rwmcintosh) [inline src/OSPSuite.Core/Chart/CurveChart.cs]: nextLegendIndex -> missing 'd'

**Pull request merged**